### ISSUE TITLE
ENS manager app sends wrong labelhash when deleting subdomains without a known preimage

### ISSUE THREAD

**Issue by narupley** [ensdomains/ens-app#1400]
**Describe the bug**
If the preimage / subdomain name is not known, the 0x labelhash shows instead on the subdomains page of the app. When you try to delete one of these subdomains from the manager, the transaction goes through successfully, but the subdomain does not actually get deleted.

**To Reproduce**
1. Have one or more subdomains without known preimages, so that it shows up as the labelhash instead:
![image](https://user-images.githubusercontent.com/3595934/146286108-9f838a35-e2c9-41a4-94f6-37d0e6ad1d2c.png)

2. Delete one of these subdomains from the manager

3. Note that the [transaction](https://etherscan.io/tx/0xaf81963f1510d17b075fcdd8cf04bc5a2a987317291dc4b8c57ffa0d117d4671) was confirmed, but the labelhash that was set is incorrect:
![image](https://user-images.githubusercontent.com/3595934/146286177-dd80cbcc-f257-4369-829b-a957435695c0.png)

In this case, the subdomain that was attempted to be deleted was:
```0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356.ethglobal.eth```
But in the [transaction](https://etherscan.io/tx/0xaf81963f1510d17b075fcdd8cf04bc5a2a987317291dc4b8c57ffa0d117d4671), the actual labelhash used was:
```0x3ca3f8d87b34d0f56a332cf6cba525e9ccca7fa084c060097982d3375df53a97```

It seems like the app is double-hashing, because:
```require('js-sha3').keccak_256('0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356');```
equals
```3ca3f8d87b34d0f56a332cf6cba525e9ccca7fa084c060097982d3375df53a97```

4. Note that after the transaction was confirmed, the subdomain still shows up on the subdomains list in the manager, as if nothing happened

**Expected behavior**
When you delete a subdomain from the subdomains list, it should be correctly deleted, regardless of whether the subdomain preimage is available or not.

**Workarounds**
You can call the setSubnodeRecord method directly with the label hash that you see on the subdomains page. However if you don't know the subdomain name itself, and you have multiple subdomains that are in this unknown state, then it's a guessing game.

**Additional context**
Multiple people on the Discord server (at least 2-3 that I'm aware of) have run into this.

**Comment by arnaud-eb** [comment]
Hi, the ENS registry contract is correctly updated but not the ens-subgraph (https://github.com/ensdomains/ens-subgraph/blob/master/src/ensRegistry.ts). Let's take the same subdomain as in your example:

`0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356.ethglobal.eth`

The `NewOwner` event was emitted and the `owner` was set to `0x000...` as expected:
![image](https://user-images.githubusercontent.com/53792888/146960038-147770d2-668b-4cd8-abba-1d68f554b48d.png)

Calling the `recordExists` function on the registry contract with the namehash as argument (`"0x52ed9364c39a1073b79c88174cce6bfb6717dbbd9075689029dc4bfaf869fce6"`) returns `true` and calling the `owner` function with that same argument returns an empty address. All good for now.
![image](https://user-images.githubusercontent.com/53792888/146960886-9c2fcc67-6247-4d0e-99e1-9b0b05471cde.png)

However, the `owner` was not reset in the ens-subgraph:
![image](https://user-images.githubusercontent.com/53792888/146961386-90428465-a0ad-443c-8da6-c1676fbf3f66.png)

This is because the namehash computed in the NewOwner event handler function takes a different label value than the one initially recorded and yields a different `id`:

`namehash(0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356.ethglobal.eth) `

`"0x52ed9364c39a1073b79c88174cce6bfb6717dbbd9075689029dc4bfaf869fce6" `as computed in the `makeSubnode` function (https://github.com/ensdomains/ens-subgraph/blob/master/src/ensRegistry.ts)

`namehash([2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356].ethglobal.eth")` 

`"0xc38aafdbf157a237513072aad73110fd33988bbf688ee489876d73e56ebbef5a"` as recorded initially in the ens-subgraph (id property).

`owner` is therefore never updated in the subgraph and the corresponding subdomain still show up in the ens manager app.

**Comment by serenae-fansubs** [comment]
If you're saying it's just a subgraph issue and the ENS registry was updated correctly, how does that explain the wrong label hash being sent in the actual smart contract transaction at https://etherscan.io/tx/0xaf81963f1510d17b075fcdd8cf04bc5a2a987317291dc4b8c57ffa0d117d4671 ?

**Comment by arnaud-eb** [comment]
Hi @serenae-fansubs , 
0x3ca3f8d87b34d0f56a332cf6cba525e9ccca7fa084c060097982d3375df53a97 is the correct labelhash. Calling the `setSubnodeRecord` method directly with 0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356 as labelhash will not update the record. Another way to check that would be to filter the `NewOwner` events by label:

0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356 will not return any transactions while 0x3ca3f8d87b34d0f56a332cf6cba525e9ccca7fa084c060097982d3375df53a97 will return 4 transactions including https://etherscan.io/tx/0xaf81963f1510d17b075fcdd8cf04bc5a2a987317291dc4b8c57ffa0d117d4671

**Comment by serenae-fansubs** [comment]
Yes, the a97 hash has related transactions now, but that's because the ENS manager UI incorrectly hashed the string `0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356` **_into_** the `0x3ca3f8d87b34d0f56a332cf6cba525e9ccca7fa084c060097982d3375df53a97` labelhash, and sent that to the user's wallet, which was then sent to the network, right?

How could the a97 labelhash possibly be correct if it's the hash of the incorrectly displayed `0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356`, and not the hash of the _actual_ preimage subdomain label?

`require('js-sha3').keccak_256('0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356');`
equals
`3ca3f8d87b34d0f56a332cf6cba525e9ccca7fa084c060097982d3375df53a97`

**Comment by arnaud-eb** [comment]
You are right. I have retrieved a record in the ens registry for the node `"0xc38aafdbf157a237513072aad73110fd33988bbf688ee489876d73e56ebbef5a"` but for some reason I don't find any event emitted for the labelhash `"0x2788f7f4d370bd792724beb1312d79af8b93c154d8d9bdf8e234c6311c05f356"`. I have just submitted a PR which might solve the deletion problem. Thanks for the feedback.
